crates/build-utils/src/dir_reader.rs

```diff
@@ -37,34 +41,45 @@ impl<'a> DirReader<'a> {
     }
 
     /// Walks the given directory
-    pub fn walk_dir(directory_path: PathWrapper) -> impl Iterator<Item = DirEntry> {
+    pub fn walk_dir(directory_path: PathWrapper) -> Vec<DirEntry> {
         WalkDir::new(Into::<PathBuf>::into(directory_path))
             .into_iter()
             .filter_map(Result::ok)
             .filter(|f| f.file_type().is_file())
+            .collect()
     }
 
-    /// Walks the given directory and stores files. If self.target is Some, it will
-    /// only store files that are in a directory from target.
+    /// Walks the given directory and stores files using par_iter.
+    /// If self.target is Some, it will only store files that
+    /// are in a directory from target.
     pub fn walk_dir_and_store_files(
-        mut self,
+        &mut self,
         directory_path: PathWrapper,
-    ) -> Result<Self, eyre::Error> {
-        for entry in Self::walk_dir(directory_path) {
-            let full_path = entry.path();
-            if let Some(target) = &self.target {
-                if target.iter().any(|t| {
-                    full_path
-                        .ancestors()
-                        .any(|a| a.to_str().map(|s| s == t).unwrap_or_default())
-                }) {
-                    continue;
+    ) -> Result<(), eyre::Error> {
+        let target = self.target.clone();
```

Comment:
Why do we need to go through this?

The fact that previous implementation was passed by value was a problem?

Reply:
This is because I now updated the implementation to iter the folders in parallel with `par_iter`, which means I need to place `self` in an `Arc<Mutex>`, which makes me lose the ownership of `self`. 
This makes it that I can't just do `self.target` anymore in the `map`, so I need to `clone` target. I realized that `target` is actually an Option which implements `Copy` so I removed the clone :)